src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/service/MappingService.kt

```diff
@@ -68,7 +68,12 @@ class MappingService(
       ?.let { RoomMappingDto(it.vsipId, it.nomisRoomDescription, it.prisonId, it.isOpen) }
       ?: throw NotFoundException("prison id=$prisonId, nomis room id=$nomisRoomDescription")
 
-  suspend fun deleteVisitMappings() = visitIdRepository.deleteAll()
+  suspend fun deleteVisitMappings(onlyMigrated: Boolean) =
+    onlyMigrated.takeIf { it }?.apply {
+      visitIdRepository.deleteByMappingTypeEquals(MappingType.MIGRATED)
+    } ?: run {
+      visitIdRepository.deleteAll()
+    }
 
```

Comment:
I wonder what we would have to do if theoretically we had say 3 parameters and we needed to build up some sort of "delete specification" 🤷🏽‍♂️

Reply:
yes in true agile style will revisit if required, I suspect JPA would have been a better choice if it got v complex